src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -329,16 +330,24 @@ void GraphOptimizer::FuseFCAndWeightsDecompression(Graph &graph) {
         }
 
         CPU_GRAPH_OPTIMIZER_SCOPE(FuseFCAndWeightsDecompression);
-        const auto multiplyConstNode = multiplyNode->getParentEdgeAt(1)->getParent();
+        const auto mulParent1 = multiplyNode->getParentEdgeAt(1)->getParent();
+        NodePtr multiplyParent, multiplyConvertNode, multiplyConstNode;
+        multiplyParent = mulParent1;
+        if (multiplyParent->getType() == Type::Convert) {
+            multiplyConvertNode = multiplyParent;
+            multiplyParent = multiplyConvertNode->getParentEdgeAt(0)->getParent();
+        }
+        multiplyConstNode = multiplyParent;
```

Comment:
these node pointer names are not helping for understanding of the pattern here:
  - `mulParent1` : produces quantization scales (in FP32 precision)
  - `multiplyConvertNode` : convert quantization scales from `fp8-e8m0` to `float` (for other quantized weight data-type like int4/int8/nf4, this convert node does npt exist - either not required or have been const-folded), this node is explicitly kept from being const-folded, so it can be fused into FC node later, in which kernel can directly access `fp8-e8m0` 
  -  `multiplyConstNode` : quantization scales  in float (or fp8-e8m0 when MXFP8/MXFP6/MXFP4 is used)

Reply:
Agree, not the bset naming. However I would prefer to avoid aby refactoring activities for this pass given the fact we will throw it out after [migration](https://github.com/openvinotoolkit/openvino/pull/25547) on corresponding ngraph based transformation.